### PR TITLE
- Added HmacSHA256 algorithm (and corresponding unit test);

### PR DESCRIPTION
**Pull request by ruzzyr**
- Added HmacSHA256 algorithm (and corresponding unit test);
- removed windows phone ifdef from on cryptographytest.cs tests

### CODE REVIEW

**Comment by keithpatton** [comment]
HMAC Algorithm Review
-------------------------

https://msdn.microsoft.com/en-us/library/system.security.cryptography.hmacsha256%28v=vs.110%29.aspx
- Ensure correct capitalization of the class name
- Ensure constructor and methods implemented match underlying .net type where possible (e.g .public constructor, no static Create). It's differently implemented to SHA1 which was copied
- Add a link to the msdn implementation as comment at top of SHA1 file for reference. 

EncryptionProvider/HashProvider Review
--------------------------

Encrypt and Decrypt do not accept a hashing algorithm, still tied to MD5, should accept HashFunctionType as param. 

Make an MD5.cs class, we should have done this earlier.

Move HashFunctionType to it's own class file.

HashProvider is not required, the individal hashing algorithm classes will be sufficient when used by themselves. 

Move HashFunctionType the HashProvider.HashString method to a private method "ComputeHashString" within EncryptionProvider and replace "HashProvider.GetMD5Hash(key)" within Encrypt/Decrypt methods to call this method. Within EncryptionProvider.ComputeHashString you can update the switch/case to ensure you use the MD5 class you have created above. 
It should also throw an ArgumentException not general exception in the default case.

In summary, we should have specific hash algorithm classes SHA1, HMACSHA256 and MD5 which mimic the underlying .net types as closely as possible. In addition we shoudl have a single EncryptionProvider class with 2 public methods Encrypt and Decrypt which accept the text to encrypt/decript, key and the algorithm choice via HashFunctionType enum. Internally, via a private method, EncryptionProvider will helpfully do the plumbing work to use the write algorithm provider to encrypt and decrypt the text which is generally all we need from it.
